packages/headless/src/features/search/search-request.ts

```diff
@@ -15,6 +16,24 @@ export const buildSearchRequest = async (state: StateNeededBySearchRequest) => {
   const sharedWithFoldingRequest =
     await buildSearchAndFoldingLoadCollectionRequest(state);
 
+  // Corner case:
+  // If the number of results requested would go over the index limit (typically 5000 results is the maximum you can reach)
+  // we need to request fewer results in order to ensure we do not receive an exception from the index
+  const ensureNumberOfResultsDoesNotGoOverIndexLimit = () => {
+    if (!state.pagination) {
```

Comment:
Shouldn't be undefined if the method is called

Reply:
TypeScript is not aware of that in the scope of this function.